### ISSUE TITLE
Object detection using OpenCV + Cascade Filter

### ISSUE THREAD

**Issue by sxppro** [Monash-Connected-Autonomous-Vehicle/AV-Competition-Software#12]
Train a cascade filter on the purple traffic lights by taking images from the Pi camera and training the filter on your own pc.

Then upload the XML file to the pi.

**Comment by AnthonyZhOon** [comment]
Tried the OpenCV cascade filter example and tutorial video https://www.youtube.com/watch?v=XrCAvs9AePM, they seem promising. Early attempts at training on the 15 or so images had some trouble, I downloaded the openCV tools for cascade filter (annotation, sampling, training) from sourceForge for the latest 3.XX version which still has the tools.
To train I needed some negative pictures with no target object that could be sampled for background to train against, so I took random pictures of my walls and floor. In the end I trained something that got stuck around 3 stages and was mostly just detecting straight lines so far.
